blockball-core/src/main/java/com/github/shynixn/blockball/core/logic/business/service/VirtualArenaServiceImpl.kt

```diff
@@ -72,20 +69,20 @@ class VirtualArenaServiceImpl @Inject constructor(
                 arena.meta.blueTeamMeta.goal.lowerCorner,
                 arena.meta.blueTeamMeta.goal.upperCorner
             )
-        })
+        }
     }
 
     /**
      * Displays the particles between the given [lowCorner] and [upCorner] location for the given [player].
      */
-    private fun displayParticles(player: Player, particle: Particle, lowCorner: Position, upCorner: Position) {
+    private fun <P> displayParticles(player: P, particle: Particle, lowCorner: Position, upCorner: Position) {
         var j = lowCorner.y
         while (j <= upCorner.y) {
             var i = lowCorner.x
             while (i <= upCorner.x) {
                 var k = lowCorner.z
                 while (k <= upCorner.z) {
-                    val location = Location(player.world, i, j, k)
+                    val location = proxyservice.getPlayerLocation<Any, P>(player)
```

Comment:
Won't this create the particles at the location of the player?

Reply:
Oh yeah, sorry. Let me work on that

Reply:
Is there any way to obtain the world of the player without referencing bukkit?

Reply:
I don't see any easy way to create a new location with specific coordinates unfortunately

Reply:
Maybe a new method for getting the world name for a player could be added to ProxyService?

Reply:
If I'm allowed to do that then sure! That should solve the problem

Reply:
Is it alright if I modify ParticleService to accept Positions(and convert them)? This would keep the dependency on Location outside the VirtualArenaService.

Reply:
@mattyway Who are you and why are you reviewing BlockBall? I do appreciate the help but I am confused.

Reply:
Hi @Shynixn I was working on this issue before I noticed that a PR had been opened. I ran into the same issue with the location.

I am just looking for projects to contribute to during hacktoberfest, and BlockBall looked interesting!

Reply:
@mattyway you can try contributing to his other project? https://github.com/Shynixn/PetBlocks/issues/298

Reply:
Hi @mattyway!
Ah, I see :) It seems like I am doing something right when so many people like to contribute to it. 

So you noticed during development that someone else has already taken your job and instead of getting annoyed by it, you have decided to help archery2000 to finish the job? That's actually amazing, I am impressed. 💯